tests/integration/test_api/test_config/test_rbac/test_rbac_mode.py

```diff
@@ -0,0 +1,183 @@
+"""
+copyright: Copyright (C) 2015-2023, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: These tests will check if the 'rbac_mode' (Role-Based Access Control) setting of the API
+       is working properly. This setting allows you to specify the operating mode between
+       'whitelist mode' and 'blacklist mode'. The Wazuh API is an open source 'RESTful' API
+       that allows for interaction with the Wazuh manager from a web browser, command line tool
+       like 'cURL' or any script or program that can make web requests.
+
+components:
+    - api
+
+suite: config
+
+targets:
+    - manager
+
+daemons:
+    - wazuh-apid
+    - wazuh-modulesd
+    - wazuh-analysisd
+    - wazuh-execd
+    - wazuh-db
+    - wazuh-remoted
+
+os_platform:
+    - linux
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/api/getting-started.html
+    - https://documentation.wazuh.com/current/user-manual/api/configuration.html#rbac-mode
+    - https://en.wikipedia.org/wiki/Role-based_access_control
+
+tags:
+    - api
+"""
+import os
+import pytest
+import requests
+
+from wazuh_testing.constants.api import CONFIGURATION_TYPES, MANAGER_INFORMATION_ROUTE
+from wazuh_testing.constants.daemons import API_DAEMONS_REQUIREMENTS
+from wazuh_testing.constants.paths.api import RBAC_DATABASE_PATH
+from wazuh_testing.modules.api.helpers import login, get_base_url
+from wazuh_testing.utils.configuration import get_test_cases_data, load_configuration_template
+from wazuh_testing.utils.database import run_sql_script
+
+
+# Marks
+pytestmark = pytest.mark.server
+
+# Variables
+# Used by add_configuration to select the target configuration file
+configuration_type = CONFIGURATION_TYPES[1]
+test_user = 'test_user'
+
+# Paths
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configuration_folder_path = os.path.join(test_data_path, 'configuration_template')
+cases_folder_path = os.path.join(test_data_path, 'test_cases')
+test_configuration_path = os.path.join(configuration_folder_path, 'configuration_rbac_mode.yaml')
+test_cases_path = os.path.join(cases_folder_path, 'cases_rbac_mode.yaml')
+
+# Configurations
+test_configuration, test_metadata, test_cases_ids = get_test_cases_data(test_cases_path)
+test_configuration = load_configuration_template(test_configuration_path, test_configuration, test_metadata)
+daemons_handler_configuration = {'daemons': API_DAEMONS_REQUIREMENTS}
+
+
+# Fixtures
```

Comment:
This has to be in a `conftest.py` file, not here.

Reply:
Same as [here](https://github.com/wazuh/wazuh/pull/17733#discussion_r1259892104)

Reply:
Done in 9afec204a9